homeassistant/components/wolflink/sensor.py

```diff
@@ -29,37 +41,90 @@ async def async_setup_entry(
-        if isinstance(parameter, ListItemParameter):
-            entities.append(WolfLinkState(coordinator, parameter, device_id))
-        if isinstance(parameter, HoursParameter):
-            entities.append(WolfLinkHours(coordinator, parameter, device_id))
-        if isinstance(parameter, SimpleParameter):
-            entities.append(WolfLinkSensor(coordinator, parameter, device_id))
+    entities: list[WolfLinkSensor] = [
+        WolfLinkSensor(
+            coordinator, parameter, device_id, get_entity_description(parameter)
+        )
+        for parameter in parameters
+    ]
 
     async_add_entities(entities, True)
 
 
+def get_entity_description(parameter: Parameter) -> SensorEntityDescription:
+    """Return the entity description for a given parameter."""
+    if isinstance(parameter, Temperature):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            device_class=SensorDeviceClass.TEMPERATURE,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        )
+    if isinstance(parameter, Pressure):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            device_class=SensorDeviceClass.PRESSURE,
+            native_unit_of_measurement=UnitOfPressure.BAR,
+        )
+    if isinstance(parameter, EnergyParameter):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            device_class=SensorDeviceClass.ENERGY,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        )
+    if isinstance(parameter, PowerParameter):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            device_class=SensorDeviceClass.POWER,
+            native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        )
+    if isinstance(parameter, PercentageParameter):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            native_unit_of_measurement=PERCENTAGE,
+        )
+    if isinstance(parameter, ListItemParameter):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            translation_key="state",
+        )
+    if isinstance(parameter, HoursParameter):
+        return SensorEntityDescription(
+            key=parameter.parameter_id,
+            name=parameter.name,
+            icon="mdi:clock",
+            native_unit_of_measurement=UnitOfTime.HOURS,
+        )
+    return SensorEntityDescription(
+        key=parameter.parameter_id,
+        name=parameter.name,
+    )
+
+
 class WolfLinkSensor(CoordinatorEntity, SensorEntity):
     """Base class for all Wolf entities."""
 
-    def __init__(self, coordinator, wolf_object: Parameter, device_id) -> None:
+    def __init__(
```

Comment:
```suggestion
    
    entity_description: WolflinkSensorEntityDescriptions
    
    def __init__(
```
should make a couple of your type checks you added unnecessary